src/core/common/notifier.cpp

```diff
@@ -233,7 +235,7 @@ void Notifier::LogEvents(Events aEvents) const
     }
 
 exit:
-    LogInfo("StateChanged (0x%08lx) %s%s]", ToUlong(aEvents.GetAsFlags()), didLog ? "... " : "[", string.AsCString());
+    LogInfo("StateChanged (0x%" PRIx64 ") %s%s]", aEvents.GetAsFlags(), didLog ? "... " : "[", string.AsCString());
```

Comment:
I strongly suggest not using `PRIx` definitions. While they are intended to help with portability across platforms and toolchains, ironically they often cause more issues than they solve. 

Some platforms/toolchains may not implement or provide them correctly. I've encountered issues in the past (in other projects) where incorrect definitions were provided by embedded toolchains, leading to subtle and hard-to-debug bugs. We faced some earlier issues in OT as well:  https://github.com/openthread/openthread/issues/5554

On a style note, I personally find them making code less readable and ugly.